users/service.go

```diff
@@ -237,10 +312,95 @@ func (svc usersService) identify(ctx context.Context, token string) (string, err
 	return email.GetValue(), nil
 }
 
+func (svc usersService) CreateGroup(ctx context.Context, token string, group Group) (Group, error) {
+	if group.Name == "" || !groupRegexp.MatchString(group.Name) {
+		return Group{}, ErrMalformedEntity
+	}
+	userID, err := svc.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return Group{}, errors.Wrap(ErrUnauthorizedAccess, err)
+	}
+	user, err := svc.users.RetrieveByEmail(ctx, userID.Value)
+	if err != nil {
+		return Group{}, errors.Wrap(ErrUnauthorizedAccess, err)
+	}
+	uid, err := uuidProvider.New().ID()
+	if err != nil {
+		return Group{}, errors.Wrap(ErrCreateUser, err)
+	}
+	group.ID = uid
+	group.OwnerID = user.ID
+	return svc.groups.Save(ctx, group)
+}
+
+func (svc usersService) ListGroups(ctx context.Context, token string, parentID string, offset, limit uint64, meta Metadata) (GroupPage, error) {
+	_, err := svc.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return GroupPage{}, errors.Wrap(ErrUnauthorizedAccess, err)
+	}
+	return svc.groups.RetrieveAll(ctx, parentID, offset, limit, meta)
+}
+
+func (svc usersService) ListGroupUsers(ctx context.Context, token, groupID string, offset, limit uint64, meta Metadata) (UserPage, error) {
+	_, err := svc.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return UserPage{}, errors.Wrap(ErrUnauthorizedAccess, err)
+	}
+	return svc.users.RetrieveAllForGroup(ctx, groupID, offset, limit, meta)
+}
+
+func (svc usersService) RemoveGroup(ctx context.Context, token, id string) error {
+	_, err := svc.auth.Identify(ctx, &mainflux.Token{Value: token})
+	if err != nil {
+		return errors.Wrap(ErrUnauthorizedAccess, err)
+	}
+	return svc.groups.Delete(ctx, id)
+}
+
+func (svc usersService) UnassignUserFromGroup(ctx context.Context, token, userID, groupID string) error {
```

Comment:
`Unassign()` would probably be enough. Maybe `UnassignMember()`, but probably just `Unassign()`